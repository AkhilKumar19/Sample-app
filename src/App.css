
html, body {
    margin: 0;
    padding: 0;
}

body {
    height: 100vh;
    width: 100vw;
}

* {
    box-sizing: border-box;
}

.App {
    text-align: center;
}

.App-logo {
    height: 40vmin;
    pointer-events: none;
}

@media (prefers-reduced-motion: no-preference) {
    .App-logo {
        animation: App-logo-spin infinite 20s linear;
    }
}

.MainComponent {
    background-color: #282c34;
    min-height: 100vh;
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
    /*font-size: calc(10px + 2vmin);*/
    color: white;
    box-sizing: border-box;
    width: 100%;
    border: solid #282c34 10px;
    padding: 5px;
}

.container-fluid{
    background-color: #282c34;
    color: white;
    min-height: 100vh;
    width: 100%;

}

.h2-font {
    font-size: calc(10px + 5vmin);
}

.App-link {
    color: #61dafb;
}

@keyframes App-logo-spin {
    from {
        transform: rotate(0deg);
    }
    to {
        transform: rotate(360deg);
    }
}
