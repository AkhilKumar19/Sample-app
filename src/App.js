import './App.css';
import {BasicStateExample} from "./components/ClassComponents/BasicStateExample/BasicStateExample";
import 'bootstrap/dist/css/bootstrap.min.css';
import 'bootstrap/dist/js/bootstrap.bundle.min';

function App() {
    let propValue = "2";
    return (
        <div className={"container-fluid text-center"}>
            <div className={"d-block h1 p-4"}>Basic React Application</div>
            <div className={"row border border-5 ml-mr-4"}>
                <BasicStateExample/>
            </div>
        </div>
    );
}

export default App;


{/*<div>*/
}
{/*    <div> Main Functional Component </div>*/
}
{/*    <>*/
}
{/*        <BasicFuncStateExample title={"Functional Component"}/>*/
}
{/*    </>*/
}
{/*</div>*/
}