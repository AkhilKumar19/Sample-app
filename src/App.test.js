import {render, screen} from '@testing-library/react';
import App from './App';

test('Snapshot Testing', () => {
  render(<App />);
  // const linkElement = screen.getByText(/learn react/i);
  // expect(linkElement).toBeInTheDocument();
  const { container } = render(<App />);
  expect(container).toMatchSnapshot();
});

test('Check If Hello Is there in the document', () => {
  render(<App />);
  // const linkElement = screen.getByText(/learn react/i);
  // expect(linkElement).toBeInTheDocument();
  const linkElement = screen.getByText("Hello");
  expect(linkElement).toBeInTheDocument();
});

test('Check If BasicComponentIsLoaded', () => {
  render(<App />);
  // const linkElement = screen.getByText(/learn react/i);
  // expect(linkElement).toBeInTheDocument();
  const linkElement = screen.getByTestId("basicComponent");
  expect(linkElement).toBeInTheDocument();
});