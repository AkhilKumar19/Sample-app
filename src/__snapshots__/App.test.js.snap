// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot Testing 1`] = `
<div>
  <div
    class="App"
    style="background-color: lightblue;"
  >
    <div>
      Main Component
    </div>
    <div>
      Hello
    </div>
    <div
      data-testid="basicComponent"
    >
      Hello Basic Component
    </div>
    <div
      style="display: flex; justify-content: center;"
    >
      <div
        style="background-color: red;"
      >
        <div>
          State Component
        </div>
        <button
          data-testid="basic-state-example-button"
        >
          counter
        </button>
      </div>
      <div
        style="background-color: lightgreen;"
      >
        <div>
          Props Component
        </div>
        <div
          class="counterMessage"
          data-testid="StateChangeMessage"
        >
          Hello 
          2
           is updated
        </div>
        <div>
          <input
            value=""
          />
          <button
            data-testid="basic-state-example-button"
          >
            counter
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
